app/controllers/main/routes.py

```diff
@@ -170,3 +172,23 @@ def searchUser(query):
 @main_bp.route('/contributors',methods = ['GET'])
 def contributors():
     return render_template("/contributors.html")
+
+
+
+@main_bp.route('/manageservicelearning', methods = ['GET'])
+def managePage():
+    """
+    This function selects all the Intructors Name and the previous courses 
+    """
+    users = User.select().where(User.isFaculty)
+    courseInstructors = CourseInstructor.select()
+
+    course_dict = {}
+
+    for instructor in courseInstructors:
+        if  (instructor.user.firstName+ " "+ instructor.user.lastName)  not in  course_dict:
+            course_dict[instructor.user.firstName+ " "+ instructor.user.lastName] =  instructor.course.courseName
+        else:
+            course_dict[instructor.user.firstName + " "+ instructor.user.lastName] += ", " + instructor.course.courseName
```

Comment:
Don't think about the formatting here in this list. That is, firstname + " " + lastname is a display consideration. Likewise, comma-separating the list of coursenames is a display detail. The view should handle that.

Also, this is a common pattern that can be shortened a bit, like we do in other places. Look at `getInstructorData` in `logic/courseManagement.py`.

Lastly, this should be a function `getAllCourseInstructors` that you can call to get the data. That way you can test it.

Reply:
`Fixed`